activestorage/lib/active_storage/service/s3_service.rb

```diff
@@ -104,6 +94,17 @@ def headers_for_direct_upload(key, content_type:, checksum:, filename: nil, disp
     end
 
     private
+      def private_url(key, expires_in:, filename:, disposition:, content_type:)
+        object_for(key).presigned_url :get, expires_in: expires_in.to_i,
+          response_content_disposition: content_disposition_with(type: disposition, filename: filename),
+          response_content_type: content_type
+      end
+
+      def public_url(key, filename, *_args)
```

Comment:
```suggestion
      def public_url(key, **)
```
To be consistent with other service overrides.